src/main/java/net/consensys/orion/http/handler/receive/ReceiveHandler.java

```diff
@@ -104,4 +99,27 @@ public void handle(final RoutingContext routingContext) {
       routingContext.response().end(toReturn);
     });
   }
+
+  private byte[] decryptPayload(
+      final RoutingContext routingContext,
+      final List<Box.PublicKey> recipients,
+      final EncryptedPayload encryptedPayload) {
+    byte[] decryptedPayload = null;
+    EnclaveException exception = null;
+    for (final Box.PublicKey recipient : recipients) {
+      try {
+        decryptedPayload = enclave.decrypt(encryptedPayload, recipient);
+      } catch (final EnclaveException e) {
+        continue;
+      }
+      // if we get here we have sucessfully decrypted the message, we can exit the loop
+      break;
+    }
+    if (decryptedPayload == null) {
```

Comment:
Think this error handling logic makes more sense in the main function this is called from

Reply:
done